packages/backend-common/src/reading/AwsS3UrlReader.ts

```diff
@@ -180,13 +185,13 @@ export class AwsS3UrlReader implements UrlReader {
       let output: ListObjectsV2Output;
       do {
         aws.config.update({ region: region });
-        output = await this.deps.s3
-          .listObjectsV2({
-            Bucket: bucket,
-            ContinuationToken: continuationToken,
-            Prefix: path,
-          })
-          .promise();
+        const request = this.deps.s3.listObjectsV2({
+          Bucket: bucket,
+          ContinuationToken: continuationToken,
+          Prefix: path,
+        });
+        options?.signal?.addEventListener('abort', () => request.abort());
```

Comment:
I'm a lil bit worried this is gonna cause a leak, but then again the V8 collector is pretty solid. Do you know that this is safe? :grin: